game-core/src/main/java/games/strategy/engine/framework/map/download/DownloadMapsWindow.java

```diff
@@ -312,134 +289,77 @@ private JTabbedPane newAvailableInstalledTabbedPanel(
+                      action,
+                      mapDownloadSwingTable::getSelectedMapNames,
+                      unsortedMaps,
+                      mapDownloadSwingTable::removeMapRow))
               .build();
       main.add(southPanel, BorderLayout.SOUTH);
     }
 
     return main;
   }
 
-  private static DownloadFileDescription determineCurrentMapSelection(
-      final List<DownloadFileDescription> maps, final Optional<String> mapToSelect) {
-    checkArgument(!maps.isEmpty());
-    if (mapToSelect.isPresent()) {
-      final Optional<DownloadFileDescription> potentialMap =
-          maps.stream().filter(m -> m.getMapName().equalsIgnoreCase(mapToSelect.get())).findFirst();
-      if (potentialMap.isPresent()) {
-        return potentialMap.get();
-      }
-    }
-
-    // just return the first map if nothing selected or could not find one
-    return maps.get(0);
-  }
-
-  private static JList<String> newGameSelectionList(
-      final DownloadFileDescription selectedMap,
-      final List<DownloadFileDescription> maps,
+  private static void newDescriptionPanelUpdatingSelectionListener(
+      final String mapName,
       final JEditorPane descriptionPanel,
-      final DefaultListModel<String> model) {
-    final JList<String> gamesList = new JList<>(model);
-    final int selectedIndex = maps.indexOf(selectedMap);
-    gamesList.setSelectedIndex(selectedIndex);
-
-    final String text = maps.get(selectedIndex).toHtmlString();
-    descriptionPanel.setText(text);
-    descriptionPanel.scrollRectToVisible(new Rectangle(0, 0, 0, 0));
-    return gamesList;
-  }
-
-  private static ListSelectionListener newDescriptionPanelUpdatingSelectionListener(
-      final JEditorPane descriptionPanel,
-      final JList<String> gamesList,
       final List<DownloadFileDescription> maps,
       final JLabel mapSizeLabelToUpdate) {
-    return e -> {
-      if (!e.getValueIsAdjusting()) {
-        final int index = gamesList.getSelectedIndex();
 
-        final boolean somethingIsSelected = index >= 0;
-        if (somethingIsSelected) {
-          final String mapName = gamesList.getModel().getElementAt(index);
-
-          // find the map description by map name and update the map download detail panel
-          maps.stream()
-              .filter(mapDescription -> mapDescription.getMapName().equals(mapName))
-              .findFirst()
-              .ifPresent(
-                  map -> {
-                    final String text = map.toHtmlString();
-                    descriptionPanel.setText(text);
-                    descriptionPanel.scrollRectToVisible(new Rectangle(0, 0, 0, 0));
-                    updateMapUrlAndSizeLabel(map, mapSizeLabelToUpdate);
-                  });
-        }
-      }
-    };
+    // find the map description by map name and update the map download detail panel
+    maps.stream()
+        .filter(mapDescription -> mapDescription.getMapName().equals(mapName))
+        .findFirst()
```

Comment:
I know this code existed before your changes, but I think there should only ever be a single element in the resulting stream, so findAny should be sufficient.
```suggestion
        .findAny()
```